ApplicationLibCode/FileInterface/RifThermalFractureReader.cpp

```diff
@@ -198,3 +213,105 @@ std::pair<QString, QString> RifThermalFractureReader::parseNameAndUnit( const QS
         return std::make_pair( "", "" );
     }
 }
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+RiaDefines::EclipseUnitSystem
+    RifThermalFractureReader::detectUnitSystem( std::shared_ptr<const RigThermalFractureDefinition> definition )
+{
+    // Use XCoord property to determine expected unit for entire file
+    QString targetName    = "XCoord";
+    auto    namesAndUnits = definition->getPropertyNamesUnits();
+    auto    res           = std::find_if( namesAndUnits.begin(), namesAndUnits.end(), [&]( const auto& val ) {
+        return val.first == targetName;
+    } );
+
+    if ( res != namesAndUnits.end() )
+    {
+        QString unit = res->second;
+        if ( unit == getExpectedUnit( targetName, RiaDefines::EclipseUnitSystem::UNITS_METRIC ) )
+            return RiaDefines::EclipseUnitSystem::UNITS_METRIC;
+        else if ( unit == getExpectedUnit( targetName, RiaDefines::EclipseUnitSystem::UNITS_FIELD ) )
+            return RiaDefines::EclipseUnitSystem::UNITS_FIELD;
+    }
+
+    return RiaDefines::EclipseUnitSystem::UNITS_UNKNOWN;
+}
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+bool RifThermalFractureReader::checkUnits( std::shared_ptr<const RigThermalFractureDefinition> definition,
+                                           RiaDefines::EclipseUnitSystem                       unitSystem )
+{
+    auto namesAndUnits = definition->getPropertyNamesUnits();
+    for ( auto [name, unit] : namesAndUnits )
+    {
+        auto expectedUnit = getExpectedUnit( name, unitSystem );
+        if ( expectedUnit != unit ) return false;
+    }
+
+    return true;
+}
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+QString RifThermalFractureReader::getExpectedUnit( const QString& name, RiaDefines::EclipseUnitSystem unitSystem )
+{
+    CAF_ASSERT( unitSystem == RiaDefines::EclipseUnitSystem::UNITS_METRIC ||
+                unitSystem == RiaDefines::EclipseUnitSystem::UNITS_FIELD );
+
+    std::map<QString, QString> mapping;
+    if ( unitSystem == RiaDefines::EclipseUnitSystem::UNITS_METRIC )
+    {
+        mapping = { { "XCoord", "m" },
```

Comment:
Consider creating a tuple with <Name>, <metric string>, <field string> to avoid duplication of variable name

Reply:
Fixed.